docs/configure-artifact-repository.md

```diff
@@ -107,43 +111,181 @@ $ cat > policy.json <<EOF
   namespace: mynamespace
 ```
 
+Use the service account in a workflow:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+spec:
+  serviceAccountName: myserviceaccount
+```
+
+#### AWS S3 with IAM Access Keys
+
+!!! Note "Least privilege user"
+    To reduce the privileges of an access key, create a user with only the necessary permissions and no more.
+
+With the bucket and policy described above, create an IAM user and add the policy:
+
+```bash
+aws iam create-user --user-name $mybucket-user
+aws iam put-user-policy --user-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
+aws iam create-access-key --user-name $mybucket-user > access-key.json
+```
+
+Configure an artifact with the access keys:
+
+```yaml
+artifacts:
+  - name: my-output-artifact
+    path: /my-output-artifact
+    s3:
+      endpoint: s3.amazonaws.com
+      bucket: my-s3-bucket
+      key: path/in/bucket/my-output-artifact.tgz
+      # The following fields are secret selectors.
+      # They reference the k8s secret named 'my-s3-credentials'.
+      # This secret is expected to have the keys 'accessKey' and 'secretKey',
+      # containing the base64 encoded credentials to the bucket.
+      accessKeySecret:
+        name: my-s3-credentials
+        key: accessKey
+      secretKeySecret:
+        name: my-s3-credentials
+        key: secretKey
+```
+
+#### AWS S3 with IAM Assume Role
+
+> v3.6 and after
+
+You can use an [IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html) for temporary access.
+
+With the bucket and policy described above, create an IAM role and add the policy:
+
+```bash
+aws iam create-role --role-name $mybucket-role
+aws iam put-role-policy --role-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
+```
+
+Retrieve the role credentials:
+
+```bash
+aws sts assume-role --role-arn arn:aws:iam::012345678901:role/$mybucket-role
+```
+
+Configure an artifact with the credentials:
+
+```yaml
+artifacts:
+  - name: my-output-artifact
+    path: /my-output-artifact
+    s3:
+      endpoint: s3.amazonaws.com
+      bucket: my-s3-bucket
+      key: path/in/bucket/my-output-artifact.tgz
+      # The following fields are secret selectors.
+      # They reference the k8s secret named 'my-s3-credentials'.
+      # This secret is expected to have the keys 'accessKey', 'secretKey', and 'sessionToken',
+      # containing the base64 encoded credentials to the bucket.
+      accessKeySecret:
+        name: my-s3-credentials
+        key: accessKey
+      secretKeySecret:
+        name: my-s3-credentials
+        key: secretKey
+      sessionTokenSecret:
+        name: my-s3-credentials
+        key: sessionToken
+```
+
+!!! Note "Temporary"
+    IAM role credentials are temporary, so you must refresh them periodically via an external mechanism.
+    Argo will not refresh them for you.
+
+### AWS S3 non-IAM Access
+
+#### AWS S3 with Access Grants
```

Comment:
```suggestion
### AWS S3 with S3 Access Grants
```

I'd actually leave out the additional "non-IAM" heading for now as there are no other methods in that same category right now. (Could also remove "non-IAM" from `.spelling` then)

Also my [previous comment](https://github.com/argoproj/argo-workflows/pull/12467#discussion_r1747482853) suggested saying "S3 Access Grants" vs. plain "Access Grants". Since S3 can be accessed via IAM and those methods specifically state "IAM" in their headings. Also "Access Grants" are very close in naming to "Access Keys" so more disambiguation is not bad in this case, IMO

Reply:
Ah yep, sorry, just misread that. Happy to change that.